frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -234,6 +243,40 @@ local footerTextGeneratorMap = {
             return T(_("%1 Off"), prefix)
         end
     end,
+    book_title = function(footer)
+        local symbol_type = footer.settings.item_prefix or "icons"
+        local prefix = symbol_prefix[symbol_type].book_title
+        local doc_info = footer.ui.document:getProps()
+        if doc_info and doc_info.title then
+            local title_text = T(_("%1 %2"), prefix, doc_info.title)
+            local face = Font:getFace(footer.text_font_face, footer.settings.text_font_size)
+            local tsize = RenderText:sizeUtf8Text(0, Screen:getWidth(), face, title_text, true, false)
+            if tsize.x > footer._saved_screen_width * footer.settings.book_title_max_width_pct / 100 then
+                title_text = RenderText:truncateTextByWidth(title_text, face,
+                    footer._saved_screen_width * footer.settings.book_title_max_width_pct / 100)
+            end
+            return title_text
+        else
+            return T(_("%1 N/A"), prefix)
+        end
+    end,
+    book_chapter = function(footer)
+        local symbol_type = footer.settings.item_prefix or "icons"
+        local prefix = symbol_prefix[symbol_type].book_chapter
+        local chapter_title = footer.ui.toc:getTocTitleByPage(footer.pageno)
+        if chapter_title then
+            local chapter_text = T(_("%1 %2"), prefix, chapter_title)
+            local face = Font:getFace(footer.text_font_face, footer.settings.text_font_size)
+            local tsize = RenderText:sizeUtf8Text(0, Screen:getWidth(), face, chapter_text, true, false)
+            if tsize.x > footer._saved_screen_width * footer.settings.book_chapter_max_width_pct / 100 then
+                chapter_text = RenderText:truncateTextByWidth(chapter_text, face,
+                    footer._saved_screen_width * footer.settings.book_chapter_max_width_pct / 100)
+            end
+            return chapter_text
+        else
+            return T(_("%1 N/A"), prefix)
```

Comment:
Or just an empty string for books with no chapter?
Or even `nil`, and have them skipped in the relevant code, to not include at all the item and its separator?